### PR TITLE
feat(amf): service reject handling changes

### PR DESCRIPTION
**Pull request by VinayKumar1591**
feat(amf): service reject handling changes

## Summary

1. Service Reject handling changes

## Test Plan

1. Tested on NGAP and Tera VM

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 1]
```diff
@@ -649,6 +649,7 @@ void amf_app_handle_pdu_session_response(
     // pdu session state
     smf_ctx->pdu_session_state = ACTIVE;
     amf_sap_t amf_sap;
+    memset(&amf_sap, 0, sizeof(amf_sap_t));
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: initialize with `amf_sap_t amf_sap = {0};`

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GGprsTimer2.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "M5GGprsTimer2.h"
+#include "M5GCommonDefs.h"
+
+namespace magma5g {
+GPRSTimer2Msg::GPRSTimer2Msg(){};
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GGprsTimer2.cpp:20]: write unit test

File: lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceReject.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#include <sstream>
+#include "M5GServiceReject.h"
+#include "M5GCommonDefs.h"
+
+namespace magma5g {
+ServiceRejectMsg::ServiceRejectMsg(){};
+ServiceRejectMsg::~ServiceRejectMsg(){};
+
+// Decoding Service Reject Message and its IEs
+int ServiceRejectMsg::DecodeServiceRejectMsg(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceReject.cpp:22]: write unit tests for encoding/decoding. i.e., encode to buffer, check success, decode the buffer, check success, check if decoded matches to the original.

**Review by ulaskozat — CHANGES_REQUESTED** [review]
add unit test. also left nit comment.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp (hunk 1 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: Done

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GGprsTimer2.cpp (hunk 2 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GGprsTimer2.cpp:20]: UT Added

File: lte/gateway/c/core/oai/test/amf/test_amf.cpp [hunk 4]
```diff
@@ -462,6 +462,30 @@ TEST(test_amf_nas5g_pkt_process, test_amf_registration_reject_msg) {
       decode_reg_rej.m5gmm_cause.m5gmm_cause);
 }
 
+TEST(test_amf_nas5g_pkt_process, test_amf_service_reject_message) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/amf/test_amf.cpp:465]: can you do this similar to the test case `test_amf_registration_reject_msg`, i.e., encode, then decode and compare the success of these ops as well as compare final decoded data is the same as the encoded one?

**Comment by ulaskozat** [comment]
please resolve conflict and also fix the unit test as per comment.

File: lte/gateway/c/core/oai/test/amf/test_amf.cpp (hunk 4 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/test/amf/test_amf.cpp:465]: Done

File: lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceReject.cpp (hunk 3 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceReject.cpp:22]: Done

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**